tests/default.nix

```diff
@@ -0,0 +1,565 @@
+{ lib, runCommand, callPackage, testcase-env }:
+
+let
+  self = {
+    # nix build .#t1.rvv-testcases.mlir.hello -L
+    mlir = {
+      axpy-masked = callPackage ./mlir/axpy-masked { };
+      conv = callPackage ./mlir/conv { };
+      hello = callPackage ./mlir/hello { };
+      matmul = callPackage ./mlir/matmul { };
+      maxvl-tail-setvl-front = callPackage ./mlir/maxvl-tail-setvl-front { };
+      rvv-vp-intrinsic-add = callPackage ./mlir/rvv-vp-intrinsic-add { };
+      rvv-vp-intrinsic-add-scalable = callPackage ./mlir/rvv-vp-intrinsic-add-scalable { };
+      stripmining = callPackage ./mlir/stripmining { };
+      vectoradd = callPackage ./mlir/vectoradd { };
+    };
+
+    # nix build .#t1.rvv-testcases.intrinsic.matmul -L
+    intrinsic = {
+      matmul = callPackage ./intrinsic/matmul { };
+      conv2d-less-m2 = callPackage ./intrinsic/conv2d_less_m2 { };
+    };
+
+    # nix build .#t1.rvv-testcases.asm.mmm -L
+    asm = {
+      fpsmoke = callPackage ./asm/fpsmoke { };
+      memcpy = callPackage ./asm/memcpy { };
+      mmm = callPackage ./asm/mmm { };
+      smoke = callPackage ./asm/smoke { };
+      strlen = callPackage ./asm/strlen { };
+      utf8-count = callPackage ./asm/utf8-count { };
+    };
+
+    # nix build .#t1.rvv-testcases.codegen.vaadd-vv -L
+    # codegen case are using xLen=32,vLen=1024 by default
+    codegen =
+      with lib;
+      let
+        # batchMkCases convert a list of name to a set of codegen case derivation.
+        # Eg. [ "vadd.vx" ] => { "vadd-vx": <vadd.vx drv> }
```

Comment:
Comment here seems a bit outdated, since the input is something like `[ { caseName = "vadd.vx"; } ]`

Reply:
Fixed.